checker/src/main/java/org/checkerframework/checker/calledmethods/builder/LombokSupport.java

```diff
@@ -0,0 +1,204 @@
+                // method called $default$foo.
+                if (methodName.startsWith("$default$")) {
+                    String propName = methodName.substring(9); // $default$ has 9 characters
+                    defaultedPropertyNames.add(propName);
+                }
+            } else if (member.getKind().isClass() && member.toString().endsWith("Builder")) {
+                // If a field bar has an @Singular annotation, Lombok always generates a method
+                // called clearBar in the builder class itself. Therefore, search the builder for
+                // such a method, and extract the appropriate property name to treat as defaulted.
+                for (Element builderMember : member.getEnclosedElements()) {
+                    if (builderMember.getKind() == ElementKind.METHOD
+                            && ElementUtils.hasAnnotation(builderMember, "lombok.Generated")) {
+                        String methodName = builderMember.getSimpleName().toString();
+                        if (methodName.startsWith("clear")) {
+                            String propName =
+                                    Introspector.decapitalize(
+                                            methodName.substring(5)); // clear has 5 characters
+                            defaultedPropertyNames.add(propName);
+                        }
+                    } else if (builderMember.getKind() == ElementKind.FIELD) {
+                        VariableTree variableTree =
+                                (VariableTree) atypeFactory.declarationFromElement(builderMember);
+                        if (variableTree != null && variableTree.getInitializer() != null) {
+                            String propName = variableTree.getName().toString();
+                            defaultedPropertyNames.add(propName);
+                            defaultedElements.put(builderMember, propName);
+                        } else if (defaultedElements.containsKey(builderMember)) {
+                            defaultedPropertyNames.add(defaultedElements.get(builderMember));
+                        }
+                    }
+                }
+            }
+        }
+        requiredPropertyNames.removeAll(defaultedPropertyNames);
+        return requiredPropertyNames;
+    }
+
+    @Override
+    public void handleConstructor(NewClassTree tree, AnnotatedTypeMirror type) {
+        return;
+    }
+
+    @Override
+    public boolean isBuilderBuildMethod(ExecutableElement element) {
+        TypeElement enclosingElement = (TypeElement) element.getEnclosingElement();
+
+        if ((ElementUtils.hasAnnotation(enclosingElement, "lombok.Generated")
+                        || ElementUtils.hasAnnotation(element, "lombok.Generated"))
+                && enclosingElement.getSimpleName().toString().endsWith("Builder")) {
+            return "build".equals(element.getSimpleName().toString());
```

Comment:
On line 199 above, the `toString` is necessary because `Name` doesn't have an `endsWith` method or equivalent.

Most of the uses of `Name#toString` precede such a call. In other cases, a mix of `Name`s and `String`s are used, so using `Name#toString` is sensible so that they can all be stored together. I don't think it's possible to remove most of the `toString` calls in this class (or even many of them), but I've removed the ones that can be replaced with `contentEquals`.